From 07b87ed8ef6c2ed144c5b646cf3116aa642fb89a Mon Sep 17 00:00:00 2001
From: Robert Hooker <sarvatt@ubuntu.com>
Date: Fri, 10 Sep 2010 07:36:12 -0400
Subject: [PATCH] Update Sandybridge device id's.

Backport of upstream commits 53767cc0d0a58d36cd445da3a31c65b349eebbba
and 104cd0554bde1d109a54db7a93700d5edfabd914. Drop after 2.12.

Signed-off-by: Robert Hooker <sarvatt@ubuntu.com>
---
 src/common.h      |   25 ++++++++++++++++++-------
 src/i810_driver.c |   23 +++++++++++++++++++++--
 src/i830_driver.c |    9 +++++++++
 3 files changed, 48 insertions(+), 9 deletions(-)

diff --git a/src/common.h b/src/common.h
index 30f1c78..6824b15 100644
--- a/src/common.h
+++ b/src/common.h
@@ -323,11 +323,17 @@ extern int I810_DEBUG;
 #define PCI_CHIP_IGDNG_M_G_BRIDGE	0x0044
 #endif
 
-#ifndef PCI_CHIP_SANDYBRIDGE
-#define PCI_CHIP_SANDYBRIDGE		0x0102
-#define PCI_CHIP_SANDYBRIDGE_BRIDGE	0x0100
-#define PCI_CHIP_SANDYBRIDGE_M		0x0106
-#define PCI_CHIP_SANDYBRIDGE_BRIDGE_M	0x0104
+#ifndef PCI_CHIP_SANDYBRIDGE_BRIDGE
+#define PCI_CHIP_SANDYBRIDGE_BRIDGE	0x0100	/* Desktop */
+#define PCI_CHIP_SANDYBRIDGE_GT1	0x0102
+#define PCI_CHIP_SANDYBRIDGE_GT2	0x0112
+#define PCI_CHIP_SANDYBRIDGE_GT2_PLUS	0x0122
+#define PCI_CHIP_SANDYBRIDGE_BRIDGE_M	0x0104	/* Mobile */
+#define PCI_CHIP_SANDYBRIDGE_M_GT1	0x0106
+#define PCI_CHIP_SANDYBRIDGE_M_GT2	0x0116
+#define PCI_CHIP_SANDYBRIDGE_M_GT2_PLUS	0x0126
+#define PCI_CHIP_SANDYBRIDGE_BRIDGE_S	0x0108	/* Server */
+#define PCI_CHIP_SANDYBRIDGE_S_GT	0x010A
 #endif
 
 #define I810_MEMBASE(p,n) (p)->regions[(n)].base_addr
@@ -385,8 +391,13 @@ extern int I810_DEBUG;
 
 #define IS_I915(pI810) (IS_I915G(pI810) || IS_I915GM(pI810) || IS_I945G(pI810) || IS_I945GM(pI810) || IS_G33CLASS(pI810))
 
-#define IS_GEN6(pI810) ((pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE || \
-			(pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE_M)
+#define IS_GEN6(pI810) ((pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE_GT1 || \
+			(pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE_GT2 || \
+			(pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE_GT2_PLUS || \
+			(pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE_M_GT1 ||\
+			(pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE_M_GT2 || \
+			(pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE_M_GT2_PLUS ||\
+			(pI810)->PciInfo->device_id == PCI_CHIP_SANDYBRIDGE_S_GT)
 
 #define IS_MOBILE(pI810) (IS_I830(pI810) || IS_I85X(pI810) || IS_I915GM(pI810) || IS_I945GM(pI810) || IS_I965GM(pI810) || IS_GM45(pI810) || IS_IGD(pI810) || IS_IGDNG_M(pI810))
 /* supports Y tiled surfaces (pre-965 Mesa isn't ready yet) */
diff --git a/src/i810_driver.c b/src/i810_driver.c
index 088b552..4440fd9 100644
--- a/src/i810_driver.c
+++ b/src/i810_driver.c
@@ -140,8 +140,13 @@ static const struct pci_id_match intel_device_match[] = {
    INTEL_DEVICE_MATCH (PCI_CHIP_B43_G, 0 ),
    INTEL_DEVICE_MATCH (PCI_CHIP_IGDNG_D_G, 0 ),
    INTEL_DEVICE_MATCH (PCI_CHIP_IGDNG_M_G, 0 ),
-   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE, 0 ),
-   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE_M, 0 ),
+   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE_GT1, 0 ),
+   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE_GT2, 0 ),
+   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE_GT2_PLUS, 0 ),
+   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE_M_GT1, 0 ),
+   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE_M_GT2, 0 ),
+   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE_M_GT2_PLUS, 0 ),
+   INTEL_DEVICE_MATCH (PCI_CHIP_SANDYBRIDGE_S_GT, 0 ),
     { 0, 0, 0 },
 };
 
@@ -196,6 +201,13 @@ static SymTabRec I810Chipsets[] = {
    {PCI_CHIP_B43_G,		"B43"},
    {PCI_CHIP_IGDNG_D_G,		"Clarkdale"},
    {PCI_CHIP_IGDNG_M_G,		"Arrandale"},
+   {PCI_CHIP_SANDYBRIDGE_GT1,	"Sandybridge" },
+   {PCI_CHIP_SANDYBRIDGE_GT2,	"Sandybridge" },
+   {PCI_CHIP_SANDYBRIDGE_GT2_PLUS,	"Sandybridge" },
+   {PCI_CHIP_SANDYBRIDGE_M_GT1,	"Sandybridge" },
+   {PCI_CHIP_SANDYBRIDGE_M_GT2,	"Sandybridge" },
+   {PCI_CHIP_SANDYBRIDGE_M_GT2_PLUS,	"Sandybridge" },
+   {PCI_CHIP_SANDYBRIDGE_S_GT,	"Sandybridge" },
    {-1,				NULL}
 };
 
@@ -235,6 +247,13 @@ static PciChipsets I810PciChipsets[] = {
    {PCI_CHIP_B43_G,		PCI_CHIP_B43_G,		NULL},
    {PCI_CHIP_IGDNG_D_G,		PCI_CHIP_IGDNG_D_G,	NULL},
    {PCI_CHIP_IGDNG_M_G,		PCI_CHIP_IGDNG_M_G,	NULL},
+   {PCI_CHIP_SANDYBRIDGE_GT1,	PCI_CHIP_SANDYBRIDGE_GT1,	NULL},
+   {PCI_CHIP_SANDYBRIDGE_GT2,	PCI_CHIP_SANDYBRIDGE_GT2,	NULL},
+   {PCI_CHIP_SANDYBRIDGE_GT2_PLUS,	PCI_CHIP_SANDYBRIDGE_GT2_PLUS,	NULL},
+   {PCI_CHIP_SANDYBRIDGE_M_GT1,	PCI_CHIP_SANDYBRIDGE_M_GT1,	NULL},
+   {PCI_CHIP_SANDYBRIDGE_M_GT2,	PCI_CHIP_SANDYBRIDGE_M_GT2,	NULL},
+   {PCI_CHIP_SANDYBRIDGE_M_GT2_PLUS,	PCI_CHIP_SANDYBRIDGE_M_GT2_PLUS, NULL},
+   {PCI_CHIP_SANDYBRIDGE_S_GT,		PCI_CHIP_SANDYBRIDGE_S_GT,	NULL},
    {-1,				-1, NULL }
 };
 
diff --git a/src/i830_driver.c b/src/i830_driver.c
index b8e0c0b..b13894e 100644
--- a/src/i830_driver.c
+++ b/src/i830_driver.c
@@ -499,6 +499,15 @@ static void i830_detect_chipset(ScrnInfoPtr scrn)
 	case PCI_CHIP_IGDNG_M_G:
 		chipname = "Arrandale";
 		break;
+	case PCI_CHIP_SANDYBRIDGE_GT1:
+	case PCI_CHIP_SANDYBRIDGE_GT2:
+	case PCI_CHIP_SANDYBRIDGE_GT2_PLUS:
+	case PCI_CHIP_SANDYBRIDGE_M_GT1:
+	case PCI_CHIP_SANDYBRIDGE_M_GT2:
+	case PCI_CHIP_SANDYBRIDGE_M_GT2_PLUS:
+	case PCI_CHIP_SANDYBRIDGE_S_GT:
+		chipname = "Sandybridge";
+		break;
 	default:
 		chipname = "unknown chipset";
 		break;
-- 
1.7.2

